core/server/worker/src/main/java/alluxio/worker/netty/DataServerBlockWriteHandler.java

```diff
@@ -67,16 +67,19 @@ public void close() throws IOException {
       } catch (Exception e) {
         throw CommonUtils.castToIOException(e);
       }
-      mWorker.cleanupSession(mSessionId);
```

Comment:
Why do we move clean up session to cancel (which I view as a valid code path) instead of close (which can be valid or invalid)?

Also, is it safe to call `commitBlock` on close since close may be called due to an error?